src/App.jsx

```diff
@@ -14,45 +16,95 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button
-        type="button"
-        className="button is-info is-light"
-      >
-        Sort alphabetically
-      </button>
-
-      <button
-        type="button"
-        className="button is-success is-light"
-      >
-        Sort by length
-      </button>
-
-      <button
-        type="button"
-        className="button is-warning is-light"
-      >
-        Reverse
-      </button>
-
-      <button
-        type="button"
-        className="button is-danger is-light"
-      >
-        Reset
-      </button>
-    </div>
+const sortByLength = 'byLength';
+const sortByAlphabet = 'ByAlphabet';
+
+function sortGoods(goods, { sortValue, reverse }) {
+  const sortedGoods = [...goods];
+
+  if (sortValue) {
+    sortedGoods.sort((good1, good2) => {
+      switch (sortValue) {
+        case sortByLength:
+          return good1.length - good2.length;
+        case sortByAlphabet:
+          return good1.localeCompare(good2);
+        default:
+          return 0;
+      }
+    });
+  }
+
+  return reverse ? sortedGoods.reverse() : sortedGoods;
+}
+
+export const App = () => {
+  const defaultSortSettings = { sortValue: '', reverse: false };
+
+  const [sortSettings, setSortSettings] = useState(defaultSortSettings);
+
+  const { sortValue, reverse } = sortSettings;
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  const goods = sortGoods(goodsFromServer, sortSettings).map(good => (
+    <li data-cy="Good" key={good}>{good}</li>
+  ));
```

Comment:
this variable looks redundant here. 
I recommends you:
1. get goods
```
const goods = sortGoods(goodsFromServer, sortSettings)
```
2. if you are familiar with `useMemo` hook you can wrap getting good with this hook
3. map sorted good at jsx below. Or you can even create a component for this list (optional)